Backpack/Calendar/Classes/BPKCalendar.m

```diff
@@ -401,6 +430,31 @@ - (void)scrollToDate:(BPKSimpleDate *)date {
     self.calendarView.pagingEnabled = NO;
 }
 
+- (void)selectWholeMonth:(BPKSimpleDate *)month {
+    self.isSelectingWholeMonth = YES;
+    NSDate *date = [month dateForCalendar:self.gregorian];
+    NSDate *minDate = [self.minDate dateForCalendar:self.gregorian];
+    NSArray<BPKSimpleDate *> *selectedDates = [self.dateProvider dateListForMonth:date fromMinDate:minDate];
+
+    BOOL previousMultiSelectionConfiguration = self.calendarView.allowsMultipleSelection;
+    self.calendarView.allowsMultipleSelection = YES;
+    [self setSelectedDates:selectedDates];
+    [self.calendarView.collectionView reloadData];
+
+    if ([self.delegate respondsToSelector:@selector(calendar:didSelectWholeMonth:)]) {
+        [self.delegate calendar:self didSelectWholeMonth:selectedDates];
+    }
+    self.calendarView.allowsMultipleSelection = previousMultiSelectionConfiguration;
+}
+
+- (BOOL)isWholeMonthButtonEnabledForMonth:(BPKSimpleDate *)month {
+    NSDate *monthDate = [month dateForCalendar:self.gregorian];
+    NSDate *minDate = [self.minDate dateForCalendar:self.gregorian];
+
+    NSDateComponents *comps = [self.gregorian components:NSCalendarUnitMonth fromDate:minDate toDate:monthDate options:0];
+    return comps.month >= 0;
+}
```

Comment:
Could we add a comment to this function that describes in what cases the wholeMonth button is enabled?

Reply:
I've added comments to all the public methods in the header file.
This the one I added for this method:
```
/**
 * Wheteher the whole month button is enabled for a given month.
 * @param month A date representing the month.
 * * @return `FALSE` if the month is previous to the `minDate`, `TRUE` otherwise.
 */
```
Let me know if you think I should add anything more or add another comment into the `BPKCalendar.m` counterpart.